### PR TITLE
Lombok + Hibernate

### PR DESCRIPTION
**Pull request by Vargisious**

### CODE REVIEW

File: src/main/java/com/practice/mappedSuperClass/entity/BaseEntity.java [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
 
 @MappedSuperclass
+@NoArgsConstructor
+@Getter
+@Setter
 public class BaseEntity {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "id")
     private int id;
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/BaseEntity.java]: Use Long

File: src/main/java/com/practice/mappedSuperClass/entity/CompanyEntity.java [hunk 2]
```diff
@@ -13,10 +18,12 @@
 
 @Entity
 @Table(name = "company")
-public class CompanyEntity extends BaseEntity {
-
-    @Column(name = "name")
-    private String name;
+@DiscriminatorValue("1")
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/CompanyEntity.java]: For what this annotation?

File: src/main/java/com/practice/mappedSuperClass/entity/ProdCompanyBaseEntity.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "prod_company_base_entity")
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+@DiscriminatorColumn(name = "type", discriminatorType = DiscriminatorType.INTEGER)
+public class ProdCompanyBaseEntity {
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/ProdCompanyBaseEntity.java]: extends BaseEntity

File: src/main/java/com/practice/mappedSuperClass/entity/ProdCompanyBaseEntity.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id")
+    int id;
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/ProdCompanyBaseEntity.java]: drop it

File: src/main/java/com/practice/mappedSuperClass/entity/ProdCompanyBaseEntity.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Column(name = "name")
+    String name;
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/ProdCompanyBaseEntity.java]: private

File: src/main/java/com/practice/mappedSuperClass/entity/CompanyEntity.java (hunk 2 above)
Comment (Vargisious) [reply src/main/java/com/practice/mappedSuperClass/entity/CompanyEntity.java]: > For what this annotation?

This is the value for my custom column "type" to differentiate between the to entities

File: src/main/java/com/practice/mappedSuperClass/entity/CompanyEntity.java [hunk 6]
```diff
@@ -12,55 +17,19 @@
...
-    @Column(name = "name")
-    private String name;
-
-    @ManyToMany(fetch = FetchType.LAZY, cascade = {CascadeType.DETACH,
-            CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH})
-    @JoinTable(name = "company_project", joinColumns = @JoinColumn(name = "company_id"),
-            inverseJoinColumns = @JoinColumn(name = "project_id"))
-    private List<ProjectEntity> projects = new ArrayList<>();
+    @Embedded
+    DeveloperCustomerEmbed developerCustomerEmbed;
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/CompanyEntity.java]: private

File: src/main/java/com/practice/mappedSuperClass/entity/CustomerEntity.java [hunk 7]
```diff
@@ -13,40 +18,18 @@
...
+@NoArgsConstructor
+@ToString
+@Getter
+@Setter
 public class CustomerEntity extends BaseEntity {
 
-    @Column(name = "name")
-    private String name;
+    @Embedded
+    DeveloperCustomerEmbed developerCustomerEmbed;
```
Comment (Aleksandr04) [inline src/main/java/com/practice/mappedSuperClass/entity/CustomerEntity.java]: private

**Pull request merged**